uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -216,7 +223,40 @@ define([
        * @return {Boolean} true is user can upload more else false
        */
       this.canUploadMore = function () {
-        return (this.request.files.length + this.request.fileUploader.queue.length) < sharedSettings.fileUploader.queueLimit;
+        var filesWithoutSoftDelete = _.filter(this.request.files, function (file) {
+          return !file.toBeDeleted;
+        });
+
+        return (filesWithoutSoftDelete.length + this.request.fileUploader.queue.length) < sharedSettings.fileUploader.queueLimit;
+      };
+
+      /**
+       * Checks if user can see file list in files tab. It needs to check for
+       * soft delete flag on leave request as well for files.
+       *
+       * @return {Boolean} true is user can see file list
+       */
+      this.canShowFileList = function () {
```

Comment:
On a second thought, it seems you need the "real" list of files (files that are there from the start + files that are being added - files that are being removed) also for deciding which messages to show.

So I would turn this as a method that returns the "real" list length (instead of it returning a boolean), and then with the returned value of the method you can decide what to do in the ui (if > 0, show the list; if = 0, show the warning message, etc)